File: src/main/java/br/com/zumbolovsky/fateapp/config/error/ErrorMessageSourceAccessor.java
package br.com.zumbolovsky.fateapp.config.error;

import org.apache.commons.lang3.ObjectUtils;
import org.springframework.context.support.MessageSourceAccessor;

import java.util.List;

public class ErrorMessageSourceAccessor {

    private final MessageSourceAccessor messageSourceAccessor;

    public ErrorMessageSourceAccessor(MessageSourceAccessor messageSourceAccessor) {
        this.messageSourceAccessor = messageSourceAccessor;
    }

    public String getMessage(String code, Object... args) {
        return messageSourceAccessor.getMessage(code, args);
    }

    public Object[] convertArgs(List<Object> args) {
        return ObjectUtils.isEmpty(args) ?
                new Object[]{} :
                args.stream()
                        .map(arg -> (arg instanceof String) ?
                                messageSourceAccessor.getMessage(arg.toString(), arg.toString()) :
                                arg)
                        .toArray();
    }

    public String getMessage(String code) {
        return messageSourceAccessor.getMessage(code);
    }
}

File: src/main/kotlin/br/com/zumbolovsky/fateapp/config/error/ErrorMessageSourceAccessor.kt
package br.com.zumbolovsky.fateapp.config.error

import org.springframework.context.support.MessageSourceAccessor

class ErrorMessageSourceAccessor(
    private val messageSourceAccessor: MessageSourceAccessor
) {

    fun getMessage(code: String, args: Array<Any>): String =
        messageSourceAccessor.getMessage(code, args)

    fun convertArgs(args: List<Any>): Array<Any> =
        if (args.isEmpty())
            emptyArray()
        else args.stream()
            .map { arg: Any ->
                if (arg is String)
                    messageSourceAccessor.getMessage(arg.toString(), arg.toString())
                else
                    arg
        }
        .toArray()

    fun getMessage(code: String): String {
        return messageSourceAccessor.getMessage(code)
    }
}